test.py

```diff
@@ -0,0 +1,98 @@
+
+class BoltTests(unittest.TestCase):
+    
+# Testing the digital write function.
+    
+    def test_1(self):
+        assert_value = json.loads(client.digitalWrite('4','HIGH'))
+        self.assertEqual(assert_value["success"],"1")
+        self.assertEqual(assert_value["value"],"1")
+        print("Digital Write Successfull!")
+
+# Testing the analog write function
+        
+    def test_3(self):
+        assert_value = json.loads(client.analogWrite('0','100'))
+        self.assertEqual(assert_value["success"],"1")
+        self.assertEqual(assert_value["value"],"1")
+        print("Analog Write Successfull!")
+        
+# Testing the digital read function.
+        
+    def test_2(self):
+        assert_value = json.loads(client.digitalRead('1'))
+        self.assertEqual(assert_value["success"],"1")
+        self.assertEqual(int(assert_value["value"]),1)
+        print("Digital Read Successfull!")
+        
+# Testing the analog read function.
+        
+    def test_4(self):
+        assert_value = json.loads(client.analogRead('A0'))
+        self.assertEqual(assert_value["success"],"1")
+        self.assertTrue(0 <= int(assert_value["value"]) <= 1024)
+        print("Analog Read Succesfull!")
+        
+# Testing the serialBegin() function.
+        
+    def test_5(self):
+        assert_value = json.loads(client.serialBegin("9600"))
+        self.assertEqual(assert_value["success"],"1")
+        self.assertEqual(assert_value["value"],"Success")
+        print("Serial Begin Successfull!")
+        
+# Testing the serialWrite() function.
+        
+    def test_6(self):
+        assert_value = json.loads(client.serialWrite('inventrom'))
+        self.assertEqual(assert_value["success"],"1")
+        self.assertEqual(assert_value["value"],"Serial write Successful")
+        print("Serial Write Successfull!")
+        
+# Testing the serialRead() function.
+        
+    def test_7(self):
+        assert_value = json.loads(client.serialRead('10'))
+        self.assertEqual(assert_value["success"],"1")
+        self.assertEqual(assert_value["value"],"inventrom")
+        print("Serial Read Successfull!")
+        
+# Testing the restart() function.
+
+    def test_restart(self):
+        assert_value = json.loads(client.restart())
+        time.sleep(5)
+        try :
+            self.assertEqual(assert_value["value"],"Restarted")
+        except AssertionError:
+            self.assertEqual(assert_value["value"], "Command timed out")
+        print("Restart Successfull!")
+        
+# Testing the isAlive() function.
+        
+    def test_8(self):
+        assert_value = json.loads(client.isAlive())
+        self.assertEqual(assert_value["success"],"1")
+        self.assertEqual(assert_value["value"],"alive")
+        print("isAlive Successfull!")
+        
+# Testing the isOnline() function
+        
+    def test_9(self):
+        assert_value = json.loads(client.isOnline())
+        self.assertEqual(str(assert_value["success"]),"1")
```

Comment:
@vimalsheoran I think you should keep the response as it is and you should make the change in your value accordingly. here you are changing the response to string. I don't think it's a good idea. Because if isonline return "1" as string there is no way we could detect that. So, I suggest you should do it like 

self.assertEqual(assert_value["success"], 1)

Reply:
The JSON reponse when unpacked gives string values. The JSON looks like `{"success":"1"}` hence in the parsed key-value pairs, both the key and the value are strings also the string value can be explicitly converted to integer (liberties of python).